libs/core/src/components/pds-select/test/pds-select.spec.tsx

```diff
@@ -0,0 +1,229 @@
+    });
+    expect(root).toEqualHtml(`
+      <pds-select component-id="field-1" label="Name">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlFor="field-1">Name</label>
+            <select class="pds-select__field" id="field-1"></select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+            <pds-icon class="pds-select__select-icon" icon="${enlarge}"></pds-icon>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('renders with error message', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" error-message="Error occurred"></pds-select>`,
+    });
+
+    const errorMessage = page.root?.shadowRoot?.querySelector('.pds-select__error-message');
+    expect(errorMessage?.textContent).toContain('Error occurred');
+  });
+
+  it('renders with helper message', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" helper-message="Helper text"></pds-select>`,
+    });
+
+    await page.waitForChanges(); // Ensures component fully renders
+
+    const helperMessage = page.root?.shadowRoot?.querySelector('.pds-select__helper-message');
+    expect(helperMessage?.textContent).toBe('Helper text');
+  });
+
+  it('renders with disabled attribute', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" disabled></pds-select>`,
+    });
+
+    await page.waitForChanges(); // Ensures component fully renders
+
+    expect(page.root).toEqualHtml(`
+      <pds-select aria-disabled="true" component-id="field-1" disabled="">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlfor="field-1"></label>
+            <select class="pds-select__field" disabled="" id="field-1">
+              <slot></slot>
+            </select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+            <pds-icon class="pds-select__select-icon" icon="${enlarge}"></pds-icon>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('updates value prop on value change', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select options='[{"value": "paul", "label": "Paul McCartney"}, {"value": "john", "label": "John Lennon"}]' />`,
+    });
+
+    const pdsSelect = page.root;
+    if (!pdsSelect || !pdsSelect.shadowRoot) {
+      throw new Error('pdsSelect or pdsSelect.shadowRoot is not available');
+    }
+    const select = pdsSelect.shadowRoot.querySelector('select');
+
+    // Change value to 'paul'
+    if (select) {
+      select.value = 'paul';
+      select.dispatchEvent(new Event('change'));
+      await page.waitForChanges();
```

Comment:
You should also be checking to see that the `pdsSelectChange` event has been emitted.

I noticed we don't have that event, this is something that should be added. As a developer, they would like the ability to hook into that event, to perform or add additional side-effects when the value changes.